File: Python3/Medium/Accepted/delete-operation-for-two-strings_5.py
class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        
        def lcs(x, y):
            m = len(x)
            n = len(y)
            dp = [[0] * (n + 1) for _ in range(m + 1)]
            for i in range(1, m + 1):
                for j in range(1, n + 1):
                    if x[i - 1] == y[j - 1]:
                        dp[i][j] = dp[i - 1][j - 1] + 1
                    else:
                        dp[i][j] = max(dp[i - 1][j], dp[i][j - 1])
            return dp[m][n]
        return len(word1) + len(word2) - 2 * lcs(word1, word2)

File: Rust/Medium/RuntimeError/delete-operation-for-two-strings_5.rs
impl Solution {
    pub fn min_distance(word1: String, word2: String) -> i32 {
        

        let (n, m) = (word1.len(), word2.len());
        let mut dp = vec![vec![0; m + 1]; n + 1];
        for i in 0..=n {
            for j in 0..=m {
                if i == 0 || j == 0 {
                    dp[i][j] = i + j;
                } else {
                    dp[i][j] = if word1.as_bytes()[i - 1] == word2.as_bytes()[j - 1] {
                        dp[i - 1][j - 1]
                    } else {
                        dp[i - 1][j - 1] + 1
                    };
                }
                dp[i][j] = dp[i][j].min(dp[i - 1][j] + 1);
                dp[i][j] = dp[i][j].min(dp[i][j - 1] + 1);
            }
        }
        dp[n][m] as i32
    }
}